### PR TITLE
fix hash checking in addon installer

### PR DESCRIPTION
**Pull request by tamland**
Fixes http://trac.kodi.tv/ticket/16973 which 'appeared' after error swallowing in GetAddonHash was removed.

(this is branched off #10817 fyi)

### CODE REVIEW

**Comment by tamland** [comment]
jenkins build this please

**Comment by tamland** [comment]
jenkins build this please

File: xbmc/filesystem/Directory.cpp [hunk 1]
```diff
@@ -390,4 +390,4 @@ void CDirectory::FilterFileDirectories(CFileItemList &items, const std::string &
         }
     }
   }
-}
+}
```
Comment (MilhouseVH) [inline xbmc/filesystem/Directory.cpp]: Can't really comment on the code although I will include it in tonight's LE test build, but can we avoid this kind of unnecessary change?

**Comment by tamland** [comment]
jenkins build this please

**Comment by MartijnKaijser** [comment]
Jenkins build this please

**Comment by MartijnKaijser** [comment]
@tamland can this go in as well or does it need some more work?

**Pull request merged**